include/deal.II/lac/relaxation_block.templates.h

```diff
@@ -94,66 +94,73 @@ inline
 void
 RelaxationBlock<MatrixType, InverseNumberType, VectorType>::invert_diagblocks ()
 {
-  const MatrixType &M=*A;
-  FullMatrix<InverseNumberType> M_cell;
-
   if (this->same_diagonal())
     {
       Assert(false, ExcNotImplemented());
     }
   else
     {
-      for (size_type block=0; block<additional_data->block_list.n_rows(); ++block)
-        {
-          const size_type bs = additional_data->block_list.row_length(block);
-          M_cell.reinit(bs, bs);
+      // compute blocks in parallel
+      Threads::TaskGroup<void> task_group;
+      for (unsigned int block = 0; block < this->additional_data->block_list.n_rows(); ++block)
```

Comment:
I think this would actually be best handled by a parallel for loop. Take a look at parallel.h. It lets each thread work on whole groups of blocks, including task stealing.